apps/src/applab/applab.js

```diff
@@ -983,6 +984,10 @@ Applab.reset = function (first) {
  */
 function runButtonClickWrapper(callback) {
   $(window).trigger('run_button_pressed');
+
+  // Change back to default screen before we serialize. This also makes it so
+  // that we'll be back on the default screen after a reset.
+  studioApp.reduxStore.dispatch(changeScreen(elementUtils.getDefaultScreenId()));
```

Comment:
So `currentScreenId` in our redux store represents the current screen in design mode. When that value changes in the store, `renderScreens` will now also update the visualization screen via `Applab.changeScreen` since we show the screen selector in code mode.